File: 백준/Bronze/10643. FUNGHI/FUNGHI.py
import sys
input = lambda: sys.stdin.readline().rstrip()


def main():
    pizza = [int(input()) for _ in range(8)]
    dap = 0
    for i in range(8):
        tmp = 0
        for j in range(4):
            idx = (i+j)%8
            tmp += pizza[idx]
        dap = max(dap, tmp)
    print(dap)


if __name__ == '__main__':
    main()

File: 백준/Bronze/10643. FUNGHI/FUNGHI.c
#include <stdio.h>

int main() {
    int pizza[8];
    for (int i = 0; i < 8; i++) {
        scanf("%d", &pizza[i]);
    }
    int dap = 0;
    for (int i = 0; i < 8; i++) {
        int tmp = 0;
        for (int j = 0; j < 4; j++) {
            int index = (i + j) % 8;
            tmp += pizza[index];
        }
        if (tmp > dap) {
            dap = tmp;
        }
    }
    printf("%d\n", dap);
    return 0;
}